articles/introduction-to-r/index.md

```diff
@@ -0,0 +1,328 @@
+   2. We use the `source` command to execute the script file of our choice. `source(file_name)` executes the entire file and gives the output in the console section. The `source(filename)` is typed and executed via the console, which we cover in the next point.
+   ![](R-Script.jpg)
+2. **Console and Terminal**: The **console** in R is similar to IDLE in Python. We can execute any command here. For this tutorial, you may use the **console**.
+   ![](console.jpg)
+3. **Environment Variables**: The **environment variables window** list all the variables created during program execution and their corresponding values. It is used to understand the dimensionality and data-types of the data being used by a program.
+   ![](environment.jpg)
+4. **Plots, Package manager, and Help Section**: The plots section generates any plots that have been specified in the R script. The **Package Manager** gives us a list of packages installed and lets us install packages if required.  Finally, the help section is a go-to for developers of all levels. The ability to go through official docs through the RStudio empowers programmers and saves them much time.
+   ![](plot.jpg)
+
+
+### R-Programming Basics
+
+In this article, we will be covering the following concepts:
+
+1. Variables, Data Types, and Operations
+2. Vectors
+3. Matrices
+4. Factors
+5. Lists
+
+#### Variables, Data Types and Operations
+
+R includes the following atomic data-types: logical, numeric, integer, complex, and character. Abstract data types such as lists, stacks, etc., are defined using these atomic data-types. Variables are memory locations that store values during the execution of a computer program. To define a variable in R, we use the reverse-arrow(<-) operator. Let us look at the code to declare variables.
+
+```R
+variable_x <- 21
+variable_y <- 54
+variable_sum <- variable_x + variable_y
+variable_complex <- 5 + 6i
+# Outputs 75
+variable_character <- "Character"
+variable_logical <- TRUE
+
+# To check the type of a variable, use class function
+class(variable_sum)
+# Output: numeric
+class(variable_character)
+# Output: character
+class(variable_logical)
+# Output: logical
+class(variable_complex)
+# Output: complex
+```
+R supports all the necessary operations such as addition, subtraction, multiplication, division, modulus, exponentiation. Complex operations are performed using these basic operations. Examples of complex operations are dot products, cross products, matrix determinants, matrix inverses, etc.
+
+#### Vectors
+Vectors are data constructs that store many elements of a particular data type. Think of them when you want to store many elements belonging to a single data-type. They are like lists in Python and are defined using the keyword `c.` Let us look at a few examples to get started with vectors:
```

Comment:
Distinguish vectors from arrays and lists. An atomic vector is different from a one-dimensional array: an array has a dim attribute of length one while a vector has no such attribute. An atomic vector is also different from a list.